gotham/src/router/mod.rs

```diff
@@ -261,6 +269,25 @@ mod tests {
         };
     }
 
+    #[test]
+    fn allow_header_if_method_not_allowed() {
+        let router = build_simple_router(|route| {
+            route.get_or_head("/test").to(handler);
```

Comment:
Worth adding a `route.get("/test").to(handler)` here as well? Will prove the dedup is occurring during finalisation.

Reply:
Added.